src/components/CommitsPage/CommitGraphWrapper.tsx

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import { ApolloError } from "@apollo/client";
+import styled from "@emotion/styled";
+import { Array } from "@ungap/global-this";
+import { Skeleton } from "antd";
+import { PageWrapper } from "components/styles";
+import { MainlineCommitsQuery } from "gql/generated/types";
+import { CommitGraph } from "./CommitGraph";
+
+export const CommitGraphWrapper: React.FC<{
+  versions: MainlineCommitsQuery["mainlineCommits"]["versions"];
+  error?: ApolloError;
+  graphType: "percentage" | "absolute";
+}> = ({ versions, error, graphType }) => {
+  let max = -1;
+  if (error) {
+    return <PageWrapper>ERROR</PageWrapper>;
+  }
+  if (!versions) {
+    return <StyledSkeleton active title={false} paragraph={{ rows: 4 }} />;
+  }
+  if (graphType === "absolute") {
+    max = findMax(data);
+  }
+  if (versions?.length !== 0) {
+    return (
+      <>
+        <FlexRowContainer>
+          {data.map((value) => (
+            <CommitGraph taskCounts={value} max={max} graphType={graphType} />
+          ))}
+        </FlexRowContainer>
+      </>
+    );
+  }
+  return <NoResults data-cy="no-patches-found">No commits found</NoResults>;
+};
+
+const StyledSkeleton = styled(Skeleton)`
+  margin-top: 12px;
+`;
+
+const FlexRowContainer = styled.div`
+  display: flex;
+  contents-direction: row;
+  justify-content: flex-start;
+  align-items: flex-end;
+  height: 222px;
+  width: 100%;
+`;
+
+const NoResults = styled.div`
+  margin-top: 12px;
+`;
+
+function findMax(data: Array<Object>) {
```

Comment:
i think something like this works also: `const findMax = (data: [{[key: string]: number}]) => data.length ? Math.max(...data.reduce((accum: [number], curr: {[key: string]: number}) => [...accum, ...Object.values(curr)], []) : -1`